File: ch2/Binary_Search/test_algorithm.cpp
#define CATCH_CONFIG_MAIN  
#include "./../../.catch.hpp"
#include "algorithm.cpp"
#include <vector>

TEST_CASE( "binary_search are computed", "[binary_search]" ) {
    std::vector<int> List = {1,5,6,8,9,10,15,19,25,29,32,40};
    REQUIRE( binary_search(List, 8) == 3 );
    REQUIRE( binary_search(List,501) == -1 );
    REQUIRE( binary_search(List,5) == 1 );
}

File: ch2/Binary_Search/test_algorithm.py
from algorithm import binary_search

def test():
    L = [1,4,7,9,10,20,25,34,90,10]
    assert binary_search(L, 25) == 6
    assert binary_search(L, 1) == 0         #===> worst case
    assert binary_search(L,123123) == -1